packages/react/src/component-tests/IcSelect/IcSelect.cy.tsx

```diff
@@ -0,0 +1,2479 @@
+    mount(
+      <IcSelect
+        label="What is your favourite coffee?"
+        options={coffeeOptionsDescriptions}
+      />
+    );
+    cy.checkHydrated("ic-select");
+    cy.clickOnShadowEl("ic-select", IC_INPUT_CONTAINER);
+    cy.checkShadowElVisible("ic-select", IC_MENU_LI);
+    cy.getWhatIsFavCoffeeQueTitle("ic-select");
+    cy.findShadowEl("ic-select", IC_MENU_LI)
+      .should(BE_VISIBLE)
+      .should(HAVE_LENGTH, "3")
+      .each(($t1) => {
+        cy.log($t1.text());
+      });
+  });
+
+  it("should render the placeholder", () => {
+    mount(
+      <IcSelect
+        label="What is your favourite coffee?"
+        options={coffeeOptionsDescriptions}
+        placeholder="Cappuccino"
+      />
+    );
+    cy.checkHydrated("ic-select");
+    cy.clickOnShadowEl("ic-select", IC_INPUT_CONTAINER);
+    cy.checkShadowElVisible("ic-select", IC_MENU_LI);
+    cy.findShadowEl("ic-select", IC_INPUT_CONTAINER)
+      .contains("Cappuccino")
+      .should(BE_VISIBLE);
+  });
+
+  it("should render as disabled correctly", () => {
+    mount(
+      <IcSelect
+        label="What is your favourite coffee?"
+        options={coffeeDisabledOption}
+        disabled
+      />
+    );
+    cy.checkHydrated("ic-select");
+    cy.get("ic-select").should(HAVE_CLASS, "disabled hydrated");
+    cy.get("ic-select").shadow().find("button").should("be.disabled");
+  });
+
+  it("should prevent click on disabled options", () => {
+    mount(
+      <IcSelect
+        label="What is your favourite coffee?"
+        options={coffeeDisabledOption}
+      />
+    );
+    cy.checkHydrated("ic-select");
+    cy.findShadowEl("ic-select", IC_INPUT_CONTAINER).type(TYPE_DOWN_ARROW);
+  });
+
+  it("should set aria-disabled and skip focus when option disabled", () => {
+    mount(
+      <IcSelect
+        label="What is your favourite coffee?"
+        options={coffeeDisabledOption}
+      />
+    );
+    cy.checkHydrated("ic-select");
+    cy.findShadowEl("ic-select", IC_INPUT_CONTAINER).type(TYPE_DOWN_ARROW);
+    cy.checkShadowElVisible("ic-select", IC_MENU_LI);
+    cy.findShadowEl("ic-select", "[aria-disabled='true']").should(
+      "not.be.focused"
+    );
+  });
+
+  it("should render as required correctly", () => {
+    mount(
+      <IcSelect
+        label="What is your favourite coffee?"
+        options={coffeeOptions}
+        required
+      />
+    );
+    cy.checkHydrated("ic-select");
+    cy.get("ic-select")
+      .shadow()
+      .contains("What is your favourite coffee? *")
+      .should(BE_VISIBLE);
+    cy.get('[required="true"]').should(BE_VISIBLE);
+  });
+
+  it("should render selected value as text when read-only", () => {
+    mount(
+      <IcSelect
+        label="What is your favourite coffee?"
+        options={coffeeOptions}
+        value="cappuccino"
+        readonly
+      />
+    );
+    cy.checkHydrated("ic-select");
+    cy.get("ic-select")
+      .shadow()
+      .contains("What is your favourite coffee? *")
```

Comment:
```suggestion
      .contains("What is your favourite coffee?")
```
won't have "*" as required prop not set